src/cocotb/_scheduler.py

```diff
@@ -514,13 +513,13 @@ async def wrapper():
     def _trigger_from_started_task(self, result: Task) -> Trigger:
         if _debug:
             self.log.debug(f"Joining to already running task: {result}")
-        return _Join(result)
+        return result.join
```

Comment:
`Scheduler._schedule_task_upon()` expects a `Trigger`, so this is failing as it returns the `Task`.